test/Ipc/ReceiverTest.php

```diff
@@ -66,4 +67,36 @@ function ($result) use (&$foo) {
         $receiver->callMessageCallback(1, 1);
         $this->assertEquals(1, $foo);
     }
+
+    /**
+     * Test Receiver jsonDecode method
+     * @expectedException ComponentException
+     */
+    public function testCallJsonDecode()
+    {
+        $app = new Application();
+        $app->setVerboseLevel(0);
+        $receiver = new Receiver($app);
+        $strJson = '{ data: "Nice data but not a valid JSON string" }';
+        $this->setExpectedException('\Gui\Exception\ComponentException');
+        $this->invokeMethod($receiver, 'jsonDecode', array($strJson));
+    }
+
+    /**
+     * Call protected/private method of a class.
+     *
+     * @param object &$object    Instantiated object that we will run method on.
+     * @param string $methodName Method name to call
+     * @param array  $parameters Array of parameters to pass into method.
+     *
+     * @return mixed Method return.
+     */
+    public function invokeMethod(&$object, $methodName, array $parameters = array())
```

Comment:
I think this method should be moved to a Util PHPUnit class to be available for the others test class